src/numpy-stubs/_typing/_ufunc.pyi

```diff
@@ -380,6 +446,113 @@ class _Call21Bool(Protocol):
         **kwds: Unpack[_Kwargs3],
     ) -> np.bool | NDArray[np.bool]: ...
 
+@type_check_only
+class _Call21Logical(Protocol):
+    @overload  # (scalar, scalar, dtype: np.object_) -> np.object_
+    def __call__(
+        self,
+        x1: _ScalarLike_co,
+        x2: _ScalarLike_co,
+        /,
+        dtype: _DTypeLike[np.object_],
+        out: None = None,
+        **kwds: Unpack[_Kwargs3],
+    ) -> bool: ...
+    @overload  # (scalar, scalar) -> bool
+    def __call__(
+        self,
+        x1: _NumberLike_co,
+        x2: _NumberLike_co,
+        /,
+        out: None = None,
+        dtype: _DTypeLikeBool | None = None,
+        **kwds: Unpack[_Kwargs3],
+    ) -> np.bool: ...
+    @overload  # (array-like, array, dtype: object_) -> Array[object_]
+    def __call__(
+        self,
+        x1: _ArrayLikeNumber_co | _ArrayLikeObject_co,
+        x2: _AnyArray,
+        /,
+        dtype: _DTypeLike[np.object_],
+        out: None = None,
+        **kwds: Unpack[_Kwargs3],
+    ) -> NDArray[np.object_]: ...
+    @overload  # (array-like, array, dtype: dtype[T]) -> Array[T]
+    def __call__(
+        self,
+        x1: _ArrayLikeNumber_co,
+        x2: NDArray[np.bool | np.number],
```

Comment:
Since this overload is the mirror image of the overload after the one below, it'd be more consistent to include `_NestedSequence` here as well

```suggestion
        x2: NDArray[np.bool | np.number] | _NestedSequence[np.bool | np.number],
```

Reply:
It might be visually helpful to swap this overload with the one below, so that the "mirror overloads" are facing each other (let's hope they get along).